vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/misc/FnDataScalarEvaluatorFactory.java

```diff
@@ -71,10 +71,12 @@ protected void evaluate(TaggedValuePointable[] args, IPointable result) throws S
                     int seqLen = seq.getEntryCount();
                     for (int j = 0; j < seqLen; ++j) {
                         seq.getEntry(j, p);
+                        check(p);
                         ah.atomize(p, ppool, tempTVP);
```

Comment:
Move the check to atomize helper class. This way any user of the atomize helper class will get the new error message.